### DISCUSSION TITLE
Recovery after netdata-meta.db corruption

### DISCUSSION THREAD

**Discussion by intelfx** [netdata/netdata#16055] [Q&A]
I have discovered that my `netdata-meta.db` had become corrupted on my main parent node. SQLite is not recognizing the database file at all, so it must be some sort of a software error, not a bit flip. The node was using the internal dbengine. Simply removing the broken database and restarting Netdata causes it to start from scratch.

Is it possible to recover the historical data (again, internal dbengine) without `netdata-meta.db`?

**Comment by stelfrag** [comment]
> Is it possible to recover the historical data (again, internal dbengine) without netdata-meta.db?

Unfortunately no. We are exploring ways to resolve this. 

If there were any errors in the logs prior to the agent failing to correctly open the database please share (it would be helpful to see if there there was an attempt to recover the database)
 
A few things:

- Is the database file in local storage?
- How many children are streaming to this parent node?
- Please share the agent version
- Please share if you have changed any options in the `[sqlite]` section in `netdata.conf`

**Reply by intelfx** [reply]
Should I keep the data directory backup around until this happens or is this a "most likely never" scenario (or will the resolution involve a disk format change, such that an old backup won't be recoverable anyway)?

> If there were any errors in the logs prior to the agent failing to correctly open the database please share (it would be helpful to see if there there was an attempt to recover the database)

I do not see any errors in the journal. Unfortunately I do not keep the `/var/log/netdata/*.log` files and they all have rotated away (in case there's some unique information being logged to the files but not the journal).

In other symptoms, it appears as if netdata just stopped writing to the database at some point in time:

```
# LC_ALL=C ls -la netdata-meta*
-rw-r----- 1 netdata netdata 219M Sep  5 12:53 netdata-meta.db
-rw-r----- 1 netdata netdata  32K Sep 26 21:29 netdata-meta.db-shm
-rw-r----- 1 netdata netdata  16M Sep  5 11:44 netdata-meta.db-wal
```

---

> Is the database file in local storage?

Yes.

> How many children are streaming to this parent node?

Just one, intermittently, and not at the moment of the corruption.

> Please share the agent version

Netdata 1.42.1 was running at the time of the corruption.

> Please share if you have changed any options in the [sqlite] section in netdata.conf

None, all defaults.

**Comment by stelfrag** [comment]
> (or will the resolution involve a disk format change, such that an old backup won't be recoverable anyway)

yes the resolution will include storing additional information to be able to make the metrics accessible.
